src/pdf_beamtime/include/pdf_beamtime/pdf_beamtime_fidpose_server.hpp

```diff
@@ -0,0 +1,60 @@
+/*Copyright 2024 Brookhaven National Laboratory
+BSD 3 Clause License. See LICENSE.txt for details.*/
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <pdf_beamtime/pdf_beamtime_server.hpp>
+#include <pdf_beamtime/tf_utilities.hpp>
+#include <pdf_beamtime_interfaces/action/fid_pose_control_msg.hpp>
+
+/// @brief Create the obstacle environment and an simple action server for the robot to move
+class PdfBeamtimeFidPoseServer : public PdfBeamtimeServer
+{
+public:
+  using FidPoseControlMsg = pdf_beamtime_interfaces::action::FidPoseControlMsg;
+
+  explicit PdfBeamtimeFidPoseServer(
+    const std::string & move_group_name, const rclcpp::NodeOptions & options,
+    std::string action_name);
+
+private:
+  moveit::core::MoveItErrorCode run_fsm(
+    std::shared_ptr<const pdf_beamtime_interfaces::action::FidPoseControlMsg_Goal> goal);
+
+  moveit::core::MoveItErrorCode run_return_fsm(
+    std::shared_ptr<const pdf_beamtime_interfaces::action::FidPoseControlMsg_Goal> goal);
```

Comment:
Do we need a different FSM for each?

Reply:
Both are the same fsm but function differently in sample place vs return.
Decided to re-structure if needed on a later PR.